### ISSUE TITLE
VO stops indicating leg position once VO statement, "You are currently on a slider...."

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/john-travoltage#440]
**Test device**
MacBook Air (M1 chip)

**Operating System**
11.4

**Browser**
safari latest

**Problem description**
https://github.com/phetsims/qa/issues/676

With VO activated, I was using the arrow keys to move the leg back and forth. Randomly, VO then states that I'm currently on a slider etc. After this statement, when I continue to move the leg back and forth with the arrow keys, VO only states the number of electrons and no longer mentions foot position. 

**Steps to reproduce**
1. Open the sim and tab into it
2. Start voice over
3. Tab over to the leg and move back and forth using the arrow keys

**Visuals**
Issue doesn't happen until ~50 seconds into the video but the entire video allows you to see the before and after.

https://drive.google.com/file/d/1t1FTtdqE37NANoQkzKU6INZ06wTZ9AxB/view?usp=sharing 

Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪John Travoltage‬
URL: https://phet-dev.colorado.edu/html/john-travoltage/1.6.0-rc.1/phet/john-travoltage_all_phet.html
Version: 1.6.0-rc.1 2021-07-22 18:05:38 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15
Language: en-us
Window: 1111x683
Pixel Ratio: 2/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0 (1.0)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 8192x8192
OES_texture_float: true
Dependencies JSON: {}

**Comment by jessegreenberg** [comment]
Thanks @Nancy-Salpepi - It looks in the video like aria-valuetext stops working for the leg after a while. I heard VO say "You are currently on a slider...." for the arm and then the the aria-valuetext worked OK. So it is either random or only happens for the leg.

Ill try to produce the problem on my Mac.

**Comment by jessegreenberg** [comment]
I am seeing something similar with my Mac with macOS 11.4. In my case, I stop getting any charge alerts or aria-valuetext when there are charges on the body. I sometimes get "Safari busy" in this case. If I move the hand/leg without any charges or right after a reset I consistently get all content. So I think there is a performance problem.

**Comment by jessegreenberg** [comment]
I also notice the fps drops way down for me when there are a large number of charges.

**Comment by jessegreenberg** [comment]
The fps is better on a built version in master, so that is a good sign.

**Comment by jessegreenberg** [comment]
This isn't really happening in the published version.

**Comment by Nancy-Salpepi** [comment]
Yeah I noticed that it wasn't happening on the published version too.

**Comment by jessegreenberg** [comment]
No worries! I actually just encountered something like this in the published version too, hearing "Safari busy" and then not much else until charges are removed from the body. When this happened for you did you hear "Safari Busy" ever @Nancy-Salpepi?

**Comment by Nancy-Salpepi** [comment]
No. I never got "safari busy."

**Comment by Nancy-Salpepi** [comment]
I can try it out if you would like.

**Comment by jessegreenberg** [comment]
No, thats OK, just curious if it was a performance issue. I just did another test where I removed all ElectronNodes from the view, and I never heard "Safari Busy", and I always heard all content as I would expect. HOWEVER, even in this case if I press and hold an arrow key I stop hearing all information about the leg position.

**Comment by jessegreenberg** [comment]
I created a basic HTML example and I am not seeing any issues yet. Here is what I tried, testing

- Including an aria-live alert
- Including the 'hairspace' workaround that AccessibleValueHandler uses.

```html
<input type="range" id="test-slider">
<label for="test-slider">My Slider</label>
<p id="alert" aria-live="polite">
</p>
```

```js
const slider = document.getElementById( "test-slider" );
const alertElement = document.getElementById( 'alert' );
let footAlert = 'Foot on rug.';
let charge = 0;
slider.addEventListener( 'keydown', change => {
  footAlert = footAlert + '\u200A';
  slider.setAttribute( 'aria-valuetext', footAlert);
  charge++;
  alertElement.textContent = `Electrons on body ${charge}`
} );
```

**Comment by jessegreenberg** [comment]
Testing the sim again, it is a bit inconsistent, and doesn't always happen. TAbbing to the arm, then back to the leg seems to fix the issue and I start hearing leg position information again. @Nancy-Salpepi is that true for you as well?

**Comment by Nancy-Salpepi** [comment]
Yes. That is the case for me as well. When I leave the leg and then go back to it, it works again.

**Comment by jessegreenberg** [comment]
OK thanks @Nancy-Salpepi.

@terracoda wondered if this could be caused by changes to the leg aria-valuetext where we are reporting the same content "foot on rug", rather than a unique string every position.

@terracoda also suggested to see if this happens in Firefox to identify where the issue is (VoiceOver vs Safari).

@jessegreenberg will test to see if a unique string each time will fix this. @terracoda will try it out on Firefox to see if it is better there.

Otherwise, since it isn't clear how to improve this issue we are OK publishing with this behavior.

**Comment by jessegreenberg** [comment]
I added an incrementing value to the end of aria-valuetext, making each string unique. I was still able to see this issue.

**Comment by jessegreenberg** [comment]
So I think that is all the time we will spend on this issue, though @terracoda will do some testing on Firefox as well. I am going to add this issue to the https://github.com/phetsims/qa/blob/master/documentation/accessibility-bugs.md.

**Comment by jessegreenberg** [comment]
Document updated above. Thanks for pointing this out @Nancy-Salpepi, sorry it doesn't seem like we can find a fix. Leaving assigned to @terracoda to review my experience on Firefox and try that out for herself.

**Comment by terracoda** [comment]
On my machine (macOS 11.4, MacBook Pro Apple M1. 16GB) using Safari and VO, I made this happen once by really pounding on the leg. I pounded less on the Arm Swing, and it didn't happen there. 

The issue was easily resolved with a Tab away and return to the Leg Swing.

I also tested on Firefox just to see how FF is doing with compatibility with VO. Using FF 90.0, VO did not read out any  of the `aria-valuetext` at all, eventhough the descriptions were indicated in the Accessibility Tree. 
<img width="967" alt="Screen Shot 2021-08-03 at 3 46 51 PM" src="https://user-images.githubusercontent.com/1902116/128068806-00a0107a-3460-468a-a94c-cea420c99ea3.png">
![Screen Shot 2021-08-03 at 4 06 40 PM](https://user-images.githubusercontent.com/1902116/128068311-d52b8d29-f2a7-4599-935e-ff24776e7865.png)

Instead of reading out the current position descriptions in the `aria-valuetext` VO just read out "0" on focus and then "0" one more time, and then context responses only after that.

Otherwise the experience with FF was quite nice. Too bad it's not yet reading out `aria-valuetext`.

I decided to test on Chrome, too. I couldn't get it to happen on Chrome. The experience on Chrome was excellent.

Considering the Leg Swing has 2 significant positions, "foot off rug" and "foot on rug", and the issue is easily resolved in Safari. This issue should not block publication.

**Comment by jessegreenberg** [comment]
Thanks @terracoda - that is exactly what I found too in Firefox. Closing this issue.
